Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------

# get the basic figure together. 

# FigureWidgets allow a figure to be dynamically updated.  We use that for two things: letting users
# change the data view with a range slider, and lettings users choose a different venue to look at from
# a dropdown menu.  

row=dfVenueKeys.iloc(0)[0]
value=row[0]

#give user option to select their venue.
venueDropdown = widgets.Dropdown(
    options=dfVenueKeys['venue_id'],
    value=value,
    description='Venue ID:',
    disabled=False,
)

container = widgets.HBox(children=[venueDropdown])

#print(venueDropdown.value)

# struggling to control the hover text properly - In Plotly express  hover_name
# property to declare what goes at the top, but that doesn't work here; not allowed. 
trace0 = go.Scatter(#customdata=dfCollatedDataSet[dfCollatedDataSet['venue_id'] == 0], 
                    y=dfCollatedDataSet[dfCollatedDataSet['venue_id']==dfVenueKeys['venue_id'].iloc(0)[0]] ['rh'], 
                    x = dfCollatedDataSet[dfCollatedDataSet['venue_id']==dfVenueKeys['venue_id'].iloc(0)[0]]['x_timestamp'], 
                    mode='lines', 
                    line_color="blue",
                    line_width = 1,
                    customdata=dfCollatedDataSet[dfCollatedDataSet['venue_id']==dfVenueKeys['venue_id'].iloc(0)[0]]['timezoneString'],

                    hovertemplate='%{y}<br>%{customdata}',
                    showlegend=False,
                    name='rh',
                    )


g = go.FigureWidget(data=trace0,
                    layout = go.Layout(
                        yaxis=dict(range=[0,100])
                    ))

print("Job Done")
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mImportError[0m                               Traceback (most recent call last)
Cell [0;32mIn[2], line 38[0m
[1;32m     20[0m [38;5;66;03m#print(venueDropdown.value)[39;00m
[1;32m     21[0m 
[1;32m     22[0m [38;5;66;03m# struggling to control the hover text properly - In Plotly express  hover_name[39;00m
[1;32m     23[0m [38;5;66;03m# property to declare what goes at the top, but that doesn't work here; not allowed. [39;00m
[1;32m     24[0m trace0 [38;5;241m=[39m go[38;5;241m.[39mScatter([38;5;66;03m#customdata=dfCollatedDataSet[dfCollatedDataSet['venue_id'] == 0], [39;00m
[1;32m     25[0m                     y[38;5;241m=[39mdfCollatedDataSet[dfCollatedDataSet[[38;5;124m'[39m[38;5;124mvenue_id[39m[38;5;124m'[39m][38;5;241m==[39mdfVenueKeys[[38;5;124m'[39m[38;5;124mvenue_id[39m[38;5;124m'[39m][38;5;241m.[39miloc([38;5;241m0[39m)[[38;5;241m0[39m]] [[38;5;124m'[39m[38;5;124mrh[39m[38;5;124m'[39m], 
[1;32m     26[0m                     x [38;5;241m=[39m dfCollatedDataSet[dfCollatedDataSet[[38;5;124m'[39m[38;5;124mvenue_id[39m[38;5;124m'[39m][38;5;241m==[39mdfVenueKeys[[38;5;124m'[39m[38;5;124mvenue_id[39m[38;5;124m'[39m][38;5;241m.[39miloc([38;5;241m0[39m)[[38;5;241m0[39m]][[38;5;124m'[39m[38;5;124mx_timestamp[39m[38;5;124m'[39m], 
[0;32m   (...)[0m
[1;32m     34[0m                     name[38;5;241m=[39m[38;5;124m'[39m[38;5;124mrh[39m[38;5;124m'[39m,
[1;32m     35[0m                     )
[0;32m---> 38[0m g [38;5;241m=[39m [43mgo[49m[38;5;241;43m.[39;49m[43mFigureWidget[49m[43m([49m[43mdata[49m[38;5;241;43m=[39;49m[43mtrace0[49m[43m,[49m
[1;32m     39[0m [43m                    [49m[43mlayout[49m[43m [49m[38;5;241;43m=[39;49m[43m [49m[43mgo[49m[38;5;241;43m.[39;49m[43mLayout[49m[43m([49m
[1;32m     40[0m [43m                        [49m[43myaxis[49m[38;5;241;43m=[39;49m[38;5;28;43mdict[39;49m[43m([49m[38;5;28;43mrange[39;49m[38;5;241;43m=[39;49m[43m[[49m[38;5;241;43m0[39;49m[43m,[49m[38;5;241;43m100[39;49m[43m][49m[43m)[49m
[1;32m     41[0m [43m                    [49m[43m)[49m[43m)[49m
[1;32m     43[0m [38;5;28mprint[39m([38;5;124m"[39m[38;5;124mJob Done[39m[38;5;124m"[39m)

File [0;32m/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/plotly/missing_anywidget.py:13[0m, in [0;36mFigureWidget.__init__[0;34m(self, *args, **kwargs)[0m
[1;32m     12[0m [38;5;28;01mdef[39;00m[38;5;250m [39m[38;5;21m__init__[39m([38;5;28mself[39m, [38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs):
[0;32m---> 13[0m     [38;5;28;01mraise[39;00m [38;5;167;01mImportError[39;00m([38;5;124m"[39m[38;5;124mPlease install anywidget to use the FigureWidget class[39m[38;5;124m"[39m)

[0;31mImportError[0m: Please install anywidget to use the FigureWidget class

